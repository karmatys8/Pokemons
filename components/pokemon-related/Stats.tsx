import React from "react";

import { ThemedView } from "../ThemedView";
import { ThemedText } from "../ThemedText";

import { Stat } from "@/types";
import { styles } from "@/assets/styles";

type StatsProps = {
  statsArray: Stat[];
};

const Stats: React.FC<StatsProps> = ({ statsArray }) => (
  <ThemedView style={styles.stepContainer}>
    <ThemedText type="subtitle">Stats</ThemedText>
    {statsArray.map((item) => (
      <ThemedText key={item.stat.name}>
        {`${item.stat.name}: ${item.base_stat}`}
      </ThemedText>
    ))}
  </ThemedView>
);

export default Stats;
