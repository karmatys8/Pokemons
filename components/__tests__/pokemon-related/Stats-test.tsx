import React from "react";
import renderer from "react-test-renderer";
import Stats from "@/components/pokemon-related/Stats";

test("renders correctly", () => {
  const statsArray = [
    {
      base_stat: 81,
      effort: 0,
      stat: { name: "hp", url: "https://pokeapi.co/api/v2/stat/1/" },
    },
    {
      base_stat: 102,
      effort: 3,
      stat: { name: "attack", url: "https://pokeapi.co/api/v2/stat/2/" },
    },
    {
      base_stat: 77,
      effort: 0,
      stat: { name: "defense", url: "https://pokeapi.co/api/v2/stat/3/" },
    },
    {
      base_stat: 85,
      effort: 0,
      stat: {
        name: "special-attack",
        url: "https://pokeapi.co/api/v2/stat/4/",
      },
    },
    {
      base_stat: 75,
      effort: 0,
      stat: {
        name: "special-defense",
        url: "https://pokeapi.co/api/v2/stat/5/",
      },
    },
    {
      base_stat: 85,
      effort: 0,
      stat: { name: "speed", url: "https://pokeapi.co/api/v2/stat/6/" },
    },
  ];

  const tree = renderer.create(<Stats statsArray={statsArray} />).toJSON();
  expect(tree).toMatchSnapshot();
});
