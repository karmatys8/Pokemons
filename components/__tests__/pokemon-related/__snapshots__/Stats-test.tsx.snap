// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "gap": 8,
        "marginBottom": 8,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        undefined,
        undefined,
        undefined,
        {
          "fontSize": 20,
          "fontWeight": "bold",
        },
        undefined,
        undefined,
      ]
    }
  >
    Stats
  </Text>
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    hp: 81
  </Text>
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    attack: 102
  </Text>
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    defense: 77
  </Text>
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    special-attack: 85
  </Text>
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    special-defense: 75
  </Text>
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    speed: 85
  </Text>
</View>
`;
